src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerImpl.java

```diff
@@ -84,83 +77,8 @@ public void schedule(
-                            new CauseAction(new BitbucketWebhookTriggerCause(triggerRequest)),
-                            triggerRequest.getAdditionalActions()));
-        }
-    }
-
-    private static class Runner implements Runnable {
-
-        private static final Action[] ACTION_ARRAY = new Action[0];
-
-        private final List<Action> actions = new ArrayList<>();
-        @Nullable
-        private final Job<?, ?> job;
-        private final SCMTriggerItem triggerItem;
-
-        private Runner(
-                @Nullable Job<?, ?> job,
-                SCMTriggerItem triggerItem,
-                Action causeAction,
-                List<Action> additionalActions) {
-            actions.add(causeAction);
-            actions.addAll(additionalActions);
-            this.triggerItem = triggerItem;
-            this.job = job;
-        }
-
-        @Override
-        public boolean equals(Object o) {
-            if (this == o) {
-                return true;
-            }
-            if (o == null || getClass() != o.getClass()) {
-                return false;
-            }
-            Runner runner = (Runner) o;
-            return Objects.equals(job, runner.job);
-        }
-
-        @Override
-        public int hashCode() {
-            return Objects.hash(job);
-        }
-
-        @Override
-        public void run() {
-            if (job == null) {
-                return;
-            }
-            File logFile = new File(job.getRootDir(), "bitbucket-webhook-trigger.log");
-            try {
-                StreamTaskListener listener = new StreamTaskListener(logFile);
-
-                long start = System.currentTimeMillis();
-                PrintStream logger = listener.getLogger();
-                logger.println(
-                        "Starting polling: "
-                                + DateFormat.getDateTimeInstance().format(new Date(start)));
-
-                PollingResult result = triggerItem.poll(listener);
-                logger.println(
-                        "Poll complete. Took "
-                                + Util.getTimeSpanString(System.currentTimeMillis() - start));
-
-                if (result.hasChanges()) {
-                    logger.println("Changes since last build: " + result.change);
-                    triggerItem.scheduleBuild2(0, actions.toArray(ACTION_ARRAY));
-                }
-            } catch (IOException | RuntimeException e) {
-                LOGGER.log(Level.SEVERE, String.format(
-                        "Failed to trigger job %s because an error occurred while writing the polling log to %s",
-                        job,
-                        logFile.getPath()),
-                        e);
-            }
+            CauseAction causeAction = new CauseAction(new BitbucketWebhookTriggerCause(triggerRequest));
+            queue.execute(new BitbucketTriggerWorker(job, triggerItem, causeAction, triggerRequest.getAdditionalActions()));
```

Comment:
nit: pass `triggerRequest` into `BitbucketTriggerWorker` and create `CauseAction` in `BitbucketTriggerWorker` constructor